NitroxModel/DataStructures/NitroxRandomStart.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Drawing;
+using NitroxModel.DataStructures.GameLogic;
+using NitroxModel.Helper;
+
+namespace NitroxModel.DataStructures
+{
+    public class NitroxRandomStart
+    {
+        private readonly Bitmap randomStartTexture;
+
+        public NitroxRandomStart(Bitmap randomStartTexture)
+        {
+            this.randomStartTexture = randomStartTexture;
+        }
+
+        public NitroxVector3 GenerateRandomStartPosition(string seed)
+        {
+
+            Random rnd = new Random(seed.GetHashCode());
+
+            for (int i = 0; i < 1000; i++)
+            {
+                float normalizedX = (float)rnd.NextDouble();
+                float normalizedZ = (float)rnd.NextDouble();
+
+                if (IsStartPointValid(normalizedX, normalizedZ))
+                {
+                    float x = 4096f * normalizedX - 2048f; // normalizedX = (x + 2048) / 4096
+                    float z = 4096f * normalizedZ - 2048f;
+                    return new NitroxVector3(x, 0, z);
+                }
+            }
+
+            return NitroxVector3.Zero;
```

Comment:
Is 0,0,0 valid?  If not, should we fall back on our current location?

Reply:
0,0,0 is the games fallback position